packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol

```diff
@@ -79,31 +79,21 @@ contract FuelMessagePortalV3 is FuelMessagePortalV2 {
      * @notice Resets the rate limit amount.
      * @param _amount The amount to reset the limit to.
      */
-    function resetRateLimitAmount(uint256 _amount) external onlyRole(SET_RATE_LIMITER_ROLE) {
-        uint256 withdrawalLimitAmountToSet;
-        bool amountWithdrawnLoweredToLimit;
-        bool withdrawalAmountResetToZero;
-        
+    function resetRateLimitAmount(uint256 _amount) external onlyRole(SET_RATE_LIMITER_ROLE) {   
+        // currentPeriodAmount is not updated when duration hasn't passed to avoid a edge case scenario where extra amount can be withdrawn from the bridge when it sshoulddn't be
+        // fo instance if rate limit is reduced & made less than the current withddrawn amount & increased to orginal before the duration ends then extra amount can be withdrawn
+        // if rate limit is reduced is reduced & made less then the current withdrawn amount then the withdraw operation from the bridge would revert until the limit is increasedd or the duration has passed
         // if period has expired then currentPeriodAmount is zero
         if (currentPeriodEnd < block.timestamp) {
             unchecked {
                 currentPeriodEnd = block.timestamp + rateLimitDuration;
             }
-            withdrawalAmountResetToZero = true;
-        } else {
-            // If the withdrawn amount is higher, it is set to the new limit amount
-            if (_amount < currentPeriodAmount) {
-                withdrawalLimitAmountToSet = _amount;
-                amountWithdrawnLoweredToLimit = true;
-            }
+
+            currentPeriodAmount = 0;
```

Comment:
can we also add a test scenario to make sure that `currentPeriodAmount` is set to zero as expected.